src/handlers/testing/align/util/session-manager.ts

```diff
@@ -0,0 +1,124 @@
+import crypto from 'crypto';
+import { writeEvaluatorTemplates } from './autogenerate';
+import { EventName, emitter } from './emitter';
+import {
+  Language,
+  type AutogeneratedEvaluator,
+  type TestCaseEvent,
+  type TestCaseResult,
+  type GradedTestCaseResult,
+} from './models';
+
+export class SessionManager {
+  private readonly apiKey: string;
+  private sessionId: string | undefined;
+  private language: Language | undefined = Language.PYTHON;
+  private testSuiteDirectory: string | undefined =
+    '/Users/nicole/autoblocks/cli';
+  private testCaseHashes: string[] = Array.from({ length: 10 }, () =>
+    crypto.randomUUID(),
+  );
+  private testCaseEvents: TestCaseEvent[] = [];
+
+  constructor(args: { apiKey: string }) {
+    this.apiKey = args.apiKey;
+  }
+
+  async start(): Promise<void> {
+    // POST /testing/local/alignment-sessions
+    this.sessionId = crypto.randomUUID();
+  }
+
+  handleInfo(args: {
+    language: Language;
+    testSuiteDirectory: string;
+    testCaseHashes: string[];
+  }) {
+    this.language = args.language;
+    this.testSuiteDirectory = args.testSuiteDirectory;
+    this.testCaseHashes = args.testCaseHashes;
+  }
+
+  getSessionId(): string {
+    if (!this.sessionId) {
+      throw new Error('Session not started');
+    }
+    return this.sessionId;
+  }
+
+  getTestCaseHashes(): string[] {
+    return this.testCaseHashes;
+  }
+
+  nextTestCaseHash(prevHash: string): string | undefined {
+    const idx = this.testCaseHashes.indexOf(prevHash);
+    if (idx !== -1 && idx < this.testCaseHashes.length - 1) {
+      return this.testCaseHashes[idx + 1];
+    }
+    return undefined;
+  }
+
+  handleTestCaseEvent(event: TestCaseEvent) {
+    this.testCaseEvents.push(event);
+  }
+
+  resetTestCaseEvents() {
+    this.testCaseEvents = [];
+  }
+
+  handleTestCaseResult(args: TestCaseResult): void {
+    emitter.emit(EventName.TEST_CASE_RESULT, args);
```

Comment:
should probably also check here that the test ID matches (and maybe that the test case hash matches the requested test case hash?)